### PR TITLE
Add state: upgrade as more intuitive alias for state: latest

### LINKED ISSUE #59: Upgrade of package when State is Present and Version is specified, fails unless forced
Offending code:

```
if (-not $package_versions.Contains($version) -and -not $allow_multiple) {
                    $module.FailJson("Chocolatey package '$package' is already installed with version(s) '$($package_versions -join "', '")' but was expecting '$version'. Either change the expected version, set state=latest, set allow_multiple=yes, or set force=yes to continue")
} elseif ($version -notin $package_versions -and $allow_multiple) {
     # add the package back into the list of missing packages if installing multiple
     $missing_packages.Add($package) > $null
}
```

WORKAROUND: 

```yaml
- name: install Notepad++ at 7.6.3
  win_chocolatey:
    name: notepadplusplus
    version: 7.6.3
    allow_prerelease: yes
    state: latest
```

### PR DESCRIPTION
**Pull request by vexx32**
## Description Of Changes

- Added `upgrade` state value as a valid option for `win_chocolatey`.
- Updated tests to also test this new state value.
- Updated py docs to make it a bit clearer why things can fail with `state: present` if a specific version is targeted

## Motivation and Context

See #59 -- our documentation and the commands necessary do not always make it clear that `state: latest` is required for certain usage patterns, especially when combined with `version:` to specify a target version.

This PR adds a `state: upgrade` as a more intuitive alias to the existing `state: latest` to better mirror the also existing `state: downgrade` and be a little clearer.

The py documentation has been updated a little to explain a little more of why playbooks that target a specific version with `state: present` may fail instead of upgrading/downgrading the package if it already exists.

## Testing

Added tests for `state: upgrade` similar to tests we have for `state: latest`

## Change Types Made

* [ ] Bug fix (non-breaking change)
* [x] Feature / Enhancement (non-breaking change)
* [ ] Breaking change (fix or feature that could cause existing functionality to change)

## Related Issue

Resolves #59

## Change Checklist

* [x] Requires a change to the documentation
* [x] Documentation has been updated
* [x] Tests to cover my changes, have been added
* [x] All new and existing tests passed.

### CODE REVIEW

**Comment by vexx32** [comment]
@corbob @JPRuskin this has been rebased on current master and should be ready for review

**Review by JPRuskin — APPROVED** [review]
LGTM!

File: build/vagrant-dependencies.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#!/bin/sh
+
+sudo add-apt-repository universe
+sudo apt update
+sudo apt install software-properties-common
```
Comment (corbob) [inline build/vagrant-dependencies.sh:5]:
```suggestion
sudo apt-get update
sudo apt-get install software-properties-common
```

When run from a script, apt complains about being unsuitable for scripts. Would it be an idea to change this to use `apt-get` instead? (Would necessitate a change to the `depenedencies.sh` as well to be consistent).

**Review by corbob — COMMENTED** [review]
A couple of questions.

File: build/Invoke-CollectionTests.ps1 [hunk 2]
```diff
@@ -35,11 +35,11 @@ begin {
 #region Bash Commands
     # All command strings in this region must be valid Bash command lines; the lines following this region join
     # the provided commands into a single command string.
-    $ImportVenv = 'source ~/ansible-venv/bin/activate'
+    $ImportVenv = '. ~/ansible-venv/bin/activate'
     $SetCollectionLocation = 'cd ~/.ansible/collections/ansible_collections/chocolatey/chocolatey'
     $SetupCommands = @(
         $ImportVenv
-        'cd ./chocolatey'
+        if ($IsCIBuild) { 'cd ./chocolatey' } else { 'cd chocolatey' }
```
[the remark this answers is not in the document]
Comment (vexx32) [reply build/Invoke-CollectionTests.ps1:42]: Ah, it should work fine. Good catch, I'll fix that up.

File: build/vagrant-dependencies.sh (hunk 1 above)
Comment (vexx32) [reply build/vagrant-dependencies.sh:5]: I can try it, unsure if it will work the same. I'll see!

Comment (vexx32) [reply build/vagrant-dependencies.sh]: Looks like it still works with apt-get 👍

**Review by corbob — DISMISSED** [review]
LGTM.

File: chocolatey/plugins/modules/win_chocolatey_source.ps1 [hunk 3]
```diff
@@ -17,11 +17,11 @@ $spec = @{
...
         bypass_proxy         = @{ type = "bool" }
         certificate          = @{ type = "str" }
-        certificate_password = @{ type = "str" }
+        certificate_password = @{ type = "str"; no_log = $true }
         priority             = @{ type = "int" }
         source               = @{ type = "str" }
         source_username      = @{ type = "str" }
-        source_password      = @{ type = "str" }
+        source_password      = @{ type = "str"; no_log = $true }
         update_password      = @{ type = "str"; default = "always"; choices = "always", "on_create" }
```
Comment (corbob) [inline chocolatey/plugins/modules/win_chocolatey_source.ps1:25]: If source_password has no_log set to true, should update_password also be set like this?

Comment (vexx32) [reply chocolatey/plugins/modules/win_chocolatey_source.ps1:25]: `update_password` isn't actually a password, it's more of a preference state that tells the module in what cases should it actually change the password if it's already set

Comment (corbob) [reply chocolatey/plugins/modules/win_chocolatey_source.ps1:25]: Oh that's kind of obvious when looking closer at the default and the choices :joy:

**Review by corbob — APPROVED** [review]
LGTM

**Pull request merged**